### ISSUE TITLE
Native markdown preview integration always uses online renderer

### ISSUE THREAD

**Issue by ChrisMuc** [qjebbs/vscode-plantuml#175]
Plantuml in markdown preview only works, when Preview Security Settings are set to "Allow insecure content". I tried to investigate why and stumbled across this: the _plantuml in markdown preview_ feature always uses the online plantuml renderer, even when plantuml render settings are set to "Local" and <empty_string>

```
User Settings:
  "plantuml.render": "Local",
  "plantuml.server": "",
```

Example markdown file (without spaces in `` `):
```
# Test

> Text

`` `plantuml
@startuml component
node app
database db
app -> db
@enduml
`` `

End
```

Preview source-code:
```html
<h1 id="test" data-line="0" class="code-line code-line" id="test-1">Test</h1>
<blockquote data-line="2" class="code-line code-line">
<p data-line="2" class="code-line code-line">Text</p>
</blockquote>

<img id="image" src="http://www.plantuml.com/plantuml/svg/SoWkIImXXXXXXXXXXXXXX-7Aoqz9LKWiAE1AIIn9XXXXXXXXXXXXXXXXXXXXXXXXXXX"><p data-line="12" class="code-line code-line">End</p>
```

**Comment by ChrisMuc** [comment]
Hi Jebbs,
sorry, it was not my intention to blame you, and I'm sure you didn't want any data to leak. I love your extension. It's awsome. By default settings are set to "Local", which is great. Now, with HTTPS, it's even more secure. And I guess these settings are really applied for your extentions preview (although not testet).
So, issue 1 can be considered as fixed. Thanks again.

Anyway, your extention integrates with vscode's internal preview for markdown to render inline puml code. And in conjunction with the internal markdown preview, the _"plantuml.render": "Local"_ setting is still ignored.
I found this line of code in _markdown-it-plantuml/render.ts_ line 15: 
`let requestUrl = plantumlServer.makeURL(diagram, format);`

I guess it's due to internal markdown preview not supporting async functions. A quick search revealed  https://github.com/markdown-it/markdown-it/blob/master/docs/development.md
The first FAQ suggests:
> **I need async rule, how to do it?**
> Sorry. You can't do it directly. All complex parsers are sync by nature. But you can use workarounds:
> 1. On parse phase, replace content by random number and store it in env.
> 2. Do async processing over collected data.
> 3. Render content and replace those random numbers with text; or replace first, then render.

So far, I didn't find the time to work on this issue and code something like the suggested solution. Maybe I'll find some time to work on this during the next days... or are you already working on a fix?

**Comment by qjebbs** [comment]
@ChrisMuc Yes, it's a potential problem, and you dig deep enough. That's what I've encountered, and I don't see a available way to bypass it. So I choose to bury my head in sand😅

If you have any idea, that would be great help.

**Comment by ChrisMuc** [comment]
As a first shot, I thought about an extra config setting, explaining the situation, and let the user decide whether he wants to use _plantuml.org_'s server for markdown-preview-integration or disabling markdown-preview-integration, when render-settings are set to _Local_.

**Comment by qjebbs** [comment]
Thanks for the advice, I'll take more time to think how to implement  a smooth as possible way for users.
